src/server/ua_services_session.c

```diff
@@ -784,8 +795,8 @@ Service_ActivateSession(UA_Server *server, UA_SecureChannel *channel,
 
         /* Check the user token signature */
         response->responseHeader.serviceResult =
-            checkSignature(server, channel->securityPolicy, tempChannelContext,
-                           &session->serverNonce, &request->userTokenSignature);
+            checkSignature(server, securityPolicy, tempChannelContext,
```

Comment:
Rename securityPolicy to utpSecurityPolicy or something else, indicating that it is the securitypolicy associated with the usertoken.